docs: improve the aws-cni chaining page

bmcustodio opened the pull request:
Improve the AWS VPC CNI plugin chaining page. Also, make use of the Cilium CLI to check for the status of the installation and for performing the connectivity test.

```release-note
docs: improve the aws-cni chaining page
```

qmonnet commented on Documentation/gettingstarted/cni-chaining-aws-cni.rst:
```diff
@@ -6,69 +6,88 @@
+
+The following command can be used to check which pods need to be restarted:
+
+.. code:: bash
```
```suggestion
.. code-block:: bash
```

qmonnet commented on Documentation/gettingstarted/cni-chaining-aws-cni.rst:
```diff
@@ -6,69 +6,88 @@
+
+   As mentioned above, some advanced Cilium features may be limited when
+   chaining with other CNI plugins. Hence, at least one of the connectivity
+   tests above may fail.
```
This reads like we expect `at least` one check to fail. In that case, we should probably name it. Otherwise, I would phrase it differently, `Hence, some of the connectivity tests above may fail`.

bmcustodio replied:
Makes total sense :+1:

qmonnet commented on Documentation/gettingstarted/k8s-install-connectivity-test.rst:
```diff
@@ -1,46 +1,23 @@
-   pod-to-b-multi-node-nodeport-7ffc76db7c-rrw82            1/1     Running   0          65s
-   pod-to-external-1111-d56f47579-d79dz                     1/1     Running   0          66s
-   pod-to-external-fqdn-allow-google-cnp-78986f4bcf-btjn7   1/1     Running   0          66s
+.. code-block:: text
```
```suggestion
::
```
(Or `.. code-block:: none` for code with no highlighting, but this is not a code snippet.) I am not entirely sure what the [`text` lexer](https://pygments.org/docs/lexers/#lexers-for-various-text-formats) refers to.

qmonnet commented on Documentation/gettingstarted/k8s-install-download-release.rst:
```diff
@@ -26,3 +21,8 @@
+.. note::
+
+   Make sure you have Helm 3 `installed <https://helm.sh/docs/intro/install/>`_.
+   Helm 2 is `no longer supported <https://helm.sh/blog/helm-v2-deprecation-timeline/>`_.
```
Why move this note? It would make sense to me before starting to run `helm`? I don't expect `helm repo add` to break with version 2, but you never know, so let's make sure people use Helm 3 from the start?

bmcustodio replied:
I actually don't remember why I moved it in the first place 🤔 I've moved it back.

qmonnet commented on Documentation/gettingstarted/k8s-install-validate.rst:
```diff
@@ -1,24 +1,30 @@
-    cilium-s8w5m                            0/1     PodInitializing     0          7s
-    coredns-86c58d9df4-4g7dd                0/1     ContainerCreating   0          8m57s
-    coredns-86c58d9df4-4l6b2                0/1     ContainerCreating   0          8m57s
+.. code::
```
```suggestion
.. code-block:: shell-session
```

qmonnet commented on Documentation/gettingstarted/k8s-install-validate.rst:
```diff
@@ -1,24 +1,30 @@
-    cilium-s8w5m                            1/1     Running   0          4m12s
-    coredns-86c58d9df4-4g7dd                1/1     Running   0          13m
-    coredns-86c58d9df4-4l6b2                1/1     Running   0          13m
+.. code::
```
```suggestion
::
```

qmonnet commented on Documentation/gettingstarted/cni-chaining-aws-cni.rst:
```diff
@@ -6,69 +6,155 @@
+
+   602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.7.10
+
+Next, patch the ``kube-system/aws-node`` ``DaemonSet`` in order to enable security
```
```suggestion
Next, patch the ``kube-system/aws-node`` DaemonSet in order to enable security
```

qmonnet reviewed (changes requested):
Contents look good to me.

On code formatting: Can you please avoid splitting the blocks? I see two consecutive `export` in section “Enabling security groups for pods (EKS)”, but I don't understand why they can't be in the same code snippet. Further down, please do not split between commands and output. It's not a bad option, but it's not consistent with what we do everywhere else in the documentation.

Regarding the directives: prefer `.. code-block::` to `.. code`. When the snippet is more about commands to type in the console, with their output, use `.. code-block:: shell-session`. The `bash` language should be for larger `bash` snippets, like the `for` loop you have to find the pods that need to be restarted. When no syntax highlighting should be performed (i.e. no command or code in the block), the marker for a literal block `::` should be used.

I understand we should document this somewhere, I started a doc but didn't have time to make a lot of progress on it.

qmonnet reviewed (approved):
Yep, looks all good this time :)

nbusseneau commented on Documentation/gettingstarted/cni-chaining-aws-cni.rst:
```diff
@@ -6,69 +6,147 @@
      --namespace kube-system \\
      --set cni.chainingMode=aws-cni \\
-     --set enableIPv4Masquerade=false \\
+     --set masquerade=false \\
```
I don't think this option has any effect: it should either be `bpf.masquerade` if you want BPF masquerading, or `enableIPv4Masquerade` / `enableIPv6Masquerade` as beforehand.

bmcustodio replied:
Nice one — I originally tested this with v1.9.x where `masquerade` is the right option. This also means that if we backport this we'll have to keep that in mind.

bmcustodio replied:
Updated!

nbusseneau commented on Documentation/gettingstarted/cni-chaining-aws-cni.rst:
```diff
@@ -6,69 +6,147 @@
+
+   As mentioned above, some advanced Cilium features may be limited when
+   chaining with other CNI plugins. Hence, some of the connectivity tests
+   above may fail.
```
In this case, I think it should be better to give the user the specific `cilium connectivity test` command [with exceptions for tests that are supposed to fail](https://github.com/cilium/cilium-cli/blob/master/internal/cli/cmd/connectivity.go#L79) with CNI chaining on AWS. For example, to skip `pod-to-local-nodeport` test:

```
cilium connectivity test --test '!pod-to-local-nodeport'
```

bmcustodio replied:
No tests seem to fail as of `1.10.0-rc1` and CLI `0.6`. I have removed the note, I guess we can always add it back if we get failing tests in the future.

nbusseneau reviewed (commented):
Thanks LGTM overall, really nicer doc :) Some nits.

bmcustodio committed:
docs: improve the aws-cni chaining page

Improve the AWS VPC CNI plugin chaining page. Also, make use of the
Cilium CLI to check for the status of the installation and for
performing the connectivity test.

Signed-off-by: Bruno Miguel Custódio <brunomcustodio@gmail.com>
